src/safe_op.hpp

```diff
@@ -223,11 +183,13 @@ T add(T summand_1, T summand_2) {
  *          when `num == std::numeric_limits<T>::min()`.
  */
 template <typename T>
-std::enable_if_t<std::is_signed_v<T>, T> abs(T num) throw() {
-  if (num == std::numeric_limits<T>::min()) {
-    return std::numeric_limits<T>::max();
+T abs(T num) noexcept {
+  if constexpr (std::is_signed_v<T>) {
+    if (num == std::numeric_limits<T>::min())
+      return std::numeric_limits<T>::max();
+    return num < 0 ? -num : num;
   }
-  return num < 0 ? -num : num;
+  return std::abs(num);
```

Comment:
The type isn't signed here so I don't think there's any need to use `std::abs`.

```suggestion
  return num;
```